Extensions/Multiplayer/multiplayerobjectruntimebehavior.ts

```diff
@@ -0,0 +1,639 @@
+  const logger = new gdjs.Logger('Multiplayer');
+  /**
+   * The MultiplayerObjectRuntimeBehavior represents a behavior that can be added to objects
+   * to make them synchronized over the network.
+   */
+  export class MultiplayerObjectRuntimeBehavior extends gdjs.RuntimeBehavior {
+    // Which player is the owner of the object.
+    // If 0, then the object is not owned by any player, so the host is the owner.
+    playerNumber: number = 0;
+
+    // The action to be executed when the player disconnects.
+    actionOnPlayerDisconnect: string;
+
+    // The last time the object has been synchronized.
+    // This is to avoid synchronizing the object too often, see _objectMaxTickRate.
+    _lastObjectSyncTimestamp: number = 0;
+    // The number of times per second the object should be synchronized if it keeps changing.
+    _objectMaxTickRate: number = 60;
+
+    // The last time the basic object info has been synchronized.
+    _lastBasicObjectSyncTimestamp: number = 0;
+    // The number of times per second the object basic info should be synchronized when it doesn't change.
+    _objectBasicInfoTickRate: number = 5;
+    // The last data sent to synchronize the basic info of the object.
+    _lastSentBasicObjectSyncData: BasicObjectNetworkSyncData | undefined;
+    // When we know that the basic info of the object has been updated, we can force sending them
+    // on the max tickrate for a number of times to ensure they are received, without the need of an acknowledgment.
+    _numberOfForcedBasicObjectUpdates: number = 0;
+
+    // The last time the variables have been synchronized.
+    _lastVariablesSyncTimestamp: number = 0;
+    // The number of times per second the variables should be synchronized.
+    _variablesTickRate: number = 1;
+    // The last data sent to synchronize the variables.
+    _lastSentVariableSyncData: VariableNetworkSyncData[] | undefined;
+    // When we know that the variables have been updated, we can force sending them
+    // on the same tickrate as the object update for a number of times
+    // to ensure they are received, without the need of an acknowledgment.
+    _numberOfForcedVariablesUpdates: number = 0;
+
+    // The last time the effects have been synchronized.
+    _lastEffectsSyncTimestamp: number = 0;
+    // The number of times per second the effects should be synchronized.
+    _effectsTickRate: number = 1;
+    // The last data sent to synchronize the effects.
+    _lastSentEffectSyncData:
+      | { [effectName: string]: EffectNetworkSyncData }
+      | undefined;
+    // When we know that the effects have been updated, we can force sending them
+    // on the same tickrate as the object update for a number of times
+    // to ensure they are received, without the need of an acknowledgment.
+    _numberOfForcedEffectsUpdates: number = 0;
+
+    // To avoid seeing too many logs.
+    _lastLogTimestamp: number = 0;
+    _logTickRate: number = 1;
+    _getTimeNow: () => number;
```

Comment:
This waste a pointer on each instance and a tiny bit of compute each time the behavior is constructed on an instance. Instead, move this into a 

```
const getTimeNow = ...
```

outside of the class :)

Reply:
I think I copied this from another place so wasn't sure if we could initiate it outside. will update